packages/playground/src/components/playground/KeyValueForm/KeyValueFormRowValue.tsx

```diff
@@ -231,3 +147,248 @@ export function KeyValueFormRowValue(props: KeyValueRowValueProps) {
+      onSubmit();
+    }
+  });
+
+  return {
+    onKeyDown,
+  };
+}
+
+function KeyValueEnum(props: {
+  value: string;
+  enumValues: Array<string>;
+  onChangeValue(value: string): void;
+}) {
+  const { enumValues, value, onChangeValue } = props;
+
+  if (enumValues.length <= 4) {
+    return (
+      <FpRadioGroup
+        defaultValue={value}
+        onValueChange={(value) => onChangeValue(value)}
+        className="flex flex-wrap gap-2"
+      >
+        {enumValues.map((enumValue) => (
+          <FpLabel
+            key={enumValue}
+            className={cn(
+              "grid grid-cols-[auto_1fr] gap-2 items-center overflow-hidden",
+              "cursor-pointer text-muted-foreground hover:text-foreground first:ml-2",
+            )}
+          >
+            <FpRadioGroupItem id={enumValue} value={enumValue} />
+            {enumValue}
+          </FpLabel>
+        ))}
+      </FpRadioGroup>
+    );
+  }
+  return (
+    <div>
+      <FpDropdownMenu>
+        <FpDropdownMenuTrigger
+          className={cn(
+            "flex",
+            "items-center",
+            "gap-2",
+            "h-min",
+            "hover:bg-muted",
+            "data-[state=open]:bg-muted",
+            "rounded-sm",
+            "w-full",
+            "group/dropdown",
+          )}
+        >
+          <div className="grow-1 w-full text-start text-sm text-muted-foreground px-2">
+            {value || "Select an option"}
+          </div>
+          <CaretSortIcon className="w-3 h-3 mr-1 flex-shrink-0 opacity-0 group-hover:opacity-100 group-focus:opacity-100 group-data-[state=open]/dropdown:opacity-100" />
+        </FpDropdownMenuTrigger>
+        <FpDropdownMenuPortal>
+          <FpDropdownMenuContent align="start">
+            <FpDropdownMenuRadioGroup value={value}>
+              {enumValues.map((enumValue) => (
+                <FpMinimalDropdownMenuRadioItem
+                  key={enumValue}
+                  onSelect={() => onChangeValue(enumValue)}
+                  value={enumValue}
+                  className={cn(
+                    "py-0.25",
+                    "focus:bg-muted",
+                    "aria-checked:bg-muted aria-checked:focus:text-accent-foreground",
+                    "cursor-pointer",
+                  )}
+                >
+                  <div
+                    className={cn(
+                      "flex-grow grid grid-cols-[2.5rem_auto] items-center gap-1 px-1 rounded-md",
+                    )}
+                  >
+                    {enumValue}
+                  </div>
+                </FpMinimalDropdownMenuRadioItem>
+              ))}
+            </FpDropdownMenuRadioGroup>
+          </FpDropdownMenuContent>
+        </FpDropdownMenuPortal>
+      </FpDropdownMenu>
+    </div>
+  );
+}
+
+function KeyValueBoolean(props: {
+  value: string;
+  onChangeValue(value: string): void;
+}) {
+  const { value, onChangeValue } = props;
+
+  return (
+    <Checkbox
```

Comment:
can we still add a `true/false` label that would mirror the state? We use checkbox for a few things in the UI so would just make it unambiguous about what kind of data this refers to

Reply:
Sorry, I somehow missed this comment, would you like to display the text "true/false" next to a checkbox?